api.rst

```diff
@@ -1509,3 +1507,5 @@ PostgREST translates `PostgreSQL error codes <https://www.postgresql.org/docs/cu
 +--------------------------+-------------------------+---------------------------------+
 | other                    | 400                     |                                 |
 +--------------------------+-------------------------+---------------------------------+
+
+.. [#] In Postgresql, unless default privileges are set otherwise, a function will begin its life with an implicit grant of the `EXECUTE` privilege to the `PUBLIC` virtual role. The term `implicit` is being used here because, for a given function, the "Access privileges" field in the output of `\df+` will start off blank even though, at this stage of the function's life, `PUBLIC` does have the function's `EXECUTE` privilege. At that time, if you make any modification to the privileges of the function, the output of the the `\df+` command will become what is conventionally expected. Notably the presence of `=X/grantor` indicates that `PUBLIC` has the `EXECUTE` privilege and, conversely, the absence of `=X/grantor` implies that `PUBLIC` does not have the `EXECUTE` privilege. If, after that, the `EXECUTE` privilege is revoked for all roles -thereby causing the "Access privileges" field in the output of `\df+` to revert to its original blank state- only the owner role (and its inheritors) will be able to execute the function. This is an inconsistency which probably should be addressed by the Postgresql developers, since the meaning of whatever `\df+` reports should (ideally) not depend on the foregoing operations.
```

Comment:
Here I think it'd be better to:

- Change `PUBLIC virtual role` to `PUBLIC role` since public is effectively a role:
```sql
create role public;
ERROR:  42939: role name "public" is reserved
LINE 1: create role public;
```
- The last part starting with `This is an inconsistency..` feels a bit like bashing on pg developers. Maybe just mention it's an inconsistency on `\df+` that users should be aware of. Ideally we should report that on a pg mailing list.